.github/CODEOWNERS

```diff
@@ -8,7 +8,8 @@
 /eng/common/                                   @dotnet/razor-tooling @dotnet/razor-compiler
 /eng/Versions.props                            @dotnet/razor-tooling @dotnet/razor-compiler
 /eng/Version.Details.xml                       @dotnet/razor-tooling @dotnet/razor-compiler
-/eng/SourceBuild*                              @dotnet/source-build-internal
+/eng/DotNetBuild.props                         @dotnet/product-construction
+/eng/SourceBuild*                              @dotnet/source-build
```

Comment:
FYI: GitHub is giving validation errors as these aliases don't have write access to these repos. Is the expectation that we need to grant such access? Wary of putting CODEOWNERS in a state where it perpetually has validation errors.

Reply:
> Is the expectation that we need to grant such access?

I think yes, per this part of the PR description:

> Repo admins, please grant @dotnet/product-construction and @dotnet/source-build write access. You can remove access granted to @dotnet/source-build-internal.